fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2994,6 +3014,19 @@ public Builder setPipelineSizer(@Nonnull PipelineSizer pipelineSizer) {
             return this;
         }
 
+        @Override
+        @Nullable
+        public Descriptors.FileDescriptor getLocalMetaDataFileDescriptor() {
+            return localMetaDataFileDescriptor;
+        }
+
+        @Override
+        @Nonnull
+        public Builder setLocalMetaDataFileDescriptor(@Nullable Descriptors.FileDescriptor localMetaDataFileDescriptor) {
+            this.localMetaDataFileDescriptor = localMetaDataFileDescriptor;
```

Comment:
I think it would be better if the meta-data store itself knew the local file descriptor so that all the record store only knew it. In fact, I'd rather the record store (and builder) only knew a meta-data provider. That requires some tweaking to the API around async and caching. But this will make it unnecessarily harder.